src/soundsourceffmpeg.cpp

```diff
@@ -531,120 +540,252 @@ unsigned int SoundSourceFFmpeg::read(unsigned long size,
 }
 
 Result SoundSourceFFmpeg::parseHeader() {
-    qDebug() << "ffmpeg: SoundSourceFFmpeg::parseHeader" << getFilename();
-    QByteArray qBAFilename = getFilename().toLocal8Bit();
-
-    AVFormatContext *FmtCtx = avformat_alloc_context();
-    AVCodecContext *CodecCtx;
-    AVDictionaryEntry *FmtTag = NULL;
-    unsigned int i;
-    AVDictionary *l_iFormatOpts = NULL;
+    QString location = getFilename();
+    setType(location.section(".",-1).toLower());
 
-    if (avformat_open_input(&FmtCtx, qBAFilename.constData(), NULL,
-                            &l_iFormatOpts) !=0) {
-        qDebug() << "av_open_input_file: cannot open" << qBAFilename.constData();
-        return ERR;
-    }
+    QByteArray qBAFilename = getFilename().toLocal8Bit();
 
-#ifndef CODEC_ID_MP3
-    if (LIBAVFORMAT_VERSION_INT > 3540580 && l_iFormatOpts != NULL) {
-        av_dict_free(&l_iFormatOpts);
-    }
-#endif
+    bool is_flac = location.endsWith("flac", Qt::CaseInsensitive);
+    bool is_wav = location.endsWith("wav", Qt::CaseInsensitive);
+    bool is_ogg = location.endsWith("ogg", Qt::CaseInsensitive);
+    bool is_mp3 = location.endsWith("mp3", Qt::CaseInsensitive);
+    bool is_mp4 = location.endsWith("mp4", Qt::CaseInsensitive) || location.endsWith("m4a", Qt::CaseInsensitive);
+    bool is_opus = location.endsWith("opus", Qt::CaseInsensitive);
+    bool is_aiff = location.endsWith("aiff", Qt::CaseInsensitive);
+
+    if (is_flac) {
+        TagLib::FLAC::File f(qBAFilename.constData());
+        if (!readFileHeader(this, f)) {
+            return ERR;
+        }
+        TagLib::Ogg::XiphComment* xiph = f.xiphComment();
+        if (xiph) {
+            readXiphComment(this, *xiph);
+        }
+        else {
+            TagLib::ID3v2::Tag *id3v2(f.ID3v2Tag());
+            if (id3v2) {
+                readID3v2Tag(this, *id3v2);
+            } else {
+                // fallback
+                const TagLib::Tag *tag(f.tag());
+                if (tag) {
+                    readTag(this, *tag);
+                } else {
+                    return ERR;
+                }
+            }
+        }
+    } else if (is_wav) {
+        TagLib::RIFF::WAV::File f(qBAFilename.constData());
+        if (!readFileHeader(this, f)) {
+            return ERR;
+        }
 
-#if LIBAVCODEC_VERSION_INT < 3622144
-    FmtCtx->max_analyze_duration = 999999999;
+        // Taglib 1.8.x doesn't provide an ID3v2Tag method for WAV files.
+#if TAGLIB_MAJOR_VERSION == 1 && TAGLIB_MINOR_VERSION == 8
```

Comment:
this test should be changed like here https://github.com/mixxxdj/mixxx/commit/6a40409b0fa5bfb53c2d3e5e7b8ab055c748ac27